framework/src/actions/PartitionerAction.C

```diff
@@ -0,0 +1,48 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "PartitionerAction.h"
+
+#include "FEProblem.h"
+#include "MooseEnum.h"
+
+#include "libmesh/linear_partitioner.h"
+
+template<>
+InputParameters validParams<PartitionerAction>()
+{
+  InputParameters params = validParams<Action>();
+  MooseEnum partitioning("linear=0", "linear");
+
+  params.addParam<MooseEnum> ("partitioner", partitioning, "The class name of the partitioner you want to use.");
+  return params;
+}
+
+PartitionerAction::PartitionerAction(InputParameters params) :
+    Action(params),
+    _partitioner_type(getParam<MooseEnum>("partitioner"))
+{
+}
+
+void
+PartitionerAction::act()
+{
+  _mesh->setIsCustomPartitionerRequested(true);
+  switch (_partitioner_type)
+  {
+    case 0:
+      _mesh->setCustomPartitioner(new LinearPartitioner());
```

Comment:
If  I use a MooseObjectAction to add partitioner, the object that is being added would need to inherit from MooseObject which is impossible for partitioner itself (or am I missing something here?). 

I would add a MoosePartitioner that inherits from MooseObject to make this thing work, but I wanted to hear (@permcody @friedmud @andrsd @YaqiWang) 's opinions about it. This would 
imply that there are two types of partitioners, the libmesh partitioner and the ones inheriting from 
MooseObject.